### PR TITLE
OPENEUROPA-1695: Implement footer block. Apply ECL component

### PR DESCRIPTION
**Pull request by sergepavle**
### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: tests/Kernel/CorporateBlocksFooterTest.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+   * @var array
+   */
+  public static $modules = [
+    'oe_corporate_blocks',
+  ];
+
+  /**
+   * Test search form block rendering.
+   */
+  public function testSearchBlockRendering(): void {
```
Comment (imanoleguskiza) [inline tests/Kernel/CorporateBlocksFooterTest.php]: Name of the function is not correct.

File: patches/@ecl/ec-component-footer+1.1.0.patch [hunk 2]
```diff
@@ -0,0 +1,22 @@
+diff --git a/node_modules/@ecl/ec-component-footer/ec-component-footer.twig b/node_modules/@ecl/ec-component-footer/ec-component-footer.twig
```
Comment (imanoleguskiza) [inline patches/@ecl/ec-component-footer+1.1.0.patch]: Did we open a PR or a ticket in ECL to fix this on their end?

File: tests/Kernel/CorporateBlocksFooterTest.php (hunk 1 above)
Comment (sergepavle) [reply tests/Kernel/CorporateBlocksFooterTest.php]: Thank you. Fixed.

File: patches/@ecl/ec-component-footer+1.1.0.patch (hunk 2 above)
Comment (sergepavle) [reply patches/@ecl/ec-component-footer+1.1.0.patch]: Created.

File: composer.json [hunk 3]
```diff
@@ -48,6 +49,10 @@
         "drupal": {
             "type": "composer",
             "url": "https://packages.drupal.org/8"
+        },
+        "oe_corporate_blocks": {
+            "url": "https://github.com/openeuropa/oe_corporate_blocks.git",
+            "type": "git"
```
Comment (ademarco) [inline composer.json]: The module is not on Packagist: https://packagist.org/packages/openeuropa/oe_corporate_blocks

File: behat.yml.dist [hunk 4]
```diff
@@ -13,16 +13,13 @@ default:
...
-              corporate top block: ".demo-block-footer-corporate-top"
-              custom block: ".demo-block-footer-custom"
+              corporate footer: "footer.ecl-footer div.ecl-footer__corporate"
               language switcher: ".ecl-language-list"
               language switcher link: ".ecl-lang-select-sites__link .ecl-lang-select-sites__label"
               language switcher overlay: ".ecl-dialog__body"
               logo: "a.ecl-logo"
               page header: ".ecl-page-header__body"
               priorities dropdown menu: "#nav-menu-expandable-group-3 .ecl-navigation-menu__links"
-              search box: ".demo-block-header-search"
```
Comment (ademarco) [inline behat.yml.dist]: This seems to be out of scope, why are we removing it here?

File: config/optional/block.block.oe_theme_corporate_footer.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    - oe_theme
+id: oe_theme_corporate_footer
+theme: oe_theme
+region: footer
+weight: 0
+provider: null
+plugin: oe_footer
+settings:
+  id: oe_footer
+  label: 'OpenEuropa Footer block'
```
Comment (ademarco) [inline config/optional/block.block.oe_theme_corporate_footer.yml]: Let's just use "Corporate footer" as label, we received instructions from COMM to try not to have OpenEuropa visible on UIs.

File: modules/oe_theme_demo/README.md [hunk 6]
```diff
@@ -10,7 +10,6 @@ The demo module includes:
 - Placeholder blocks like:
  - Language switcher
  - Site switcher
- - Search block
  - Footer
```
Comment (ademarco) [inline modules/oe_theme_demo/README.md]: The footer is also not provided by the demo module anymore right?

File: oe_theme.theme [hunk 7]
```diff
@@ -795,7 +796,23 @@ function oe_theme_preprocess_pattern_footer(array &$variables): void {
  */
 function _oe_theme_preprocess_link_types(array &$links): void {
   foreach ($links as &$link) {
-    $link['type'] = $link['link_type'];
-    unset($link['link_type']);
+    if (isset($link['link_type'])) {
+      $link['type'] = $link['link_type'];
+      unset($link['link_type']);
```
Comment (ademarco) [inline oe_theme.theme]: We don't need this anymore after decision taken on #185, i.e. the footer is not an UI pattern anymore.

File: oe_theme.theme [hunk 8]
```diff
@@ -795,7 +796,23 @@ function oe_theme_preprocess_pattern_footer(array &$variables): void {
...
+    }
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function oe_theme_preprocess_corporate_blocks_footer(array &$variables): void {
+  $variables['content'] = [
+    '#type' => 'pattern',
```
Comment (ademarco) [inline oe_theme.theme]: We don't need this preprocess anymore after decision taken on #185, i.e. the footer is not an UI pattern anymore.

File: composer.json (hunk 3 above)
Comment (sergepavle) [reply composer.json]: Fixed.

File: behat.yml.dist (hunk 4 above)
Comment (sergepavle) [reply behat.yml.dist]: It looks like this PR missed attention: https://github.com/openeuropa/oe_theme/pull/176
So I did the same within this PR.

File: config/optional/block.block.oe_theme_corporate_footer.yml (hunk 5 above)
Comment (sergepavle) [reply config/optional/block.block.oe_theme_corporate_footer.yml]: fixed

File: oe_theme.theme (hunk 7 above)
Comment (sergepavle) [reply oe_theme.theme]: Done

File: modules/oe_theme_demo/README.md (hunk 6 above)
Comment (sergepavle) [reply modules/oe_theme_demo/README.md]: yes. Fixed.

File: oe_theme.theme (hunk 8 above)
Comment (sergepavle) [reply oe_theme.theme]: Done

File: templates/navigation/corporate-blocks-footer.html.twig [hunk 9]
```diff
@@ -0,0 +1,15 @@
+{#
+/**
+ * @file
+ * Theme override for the corporate block footer.
+ *
+ * Available variables:
+ * - content: rendered footer pattern. *
```
Comment (ademarco) [inline templates/navigation/corporate-blocks-footer.html.twig]: Small typo at end of line ` * - content: rendered footer pattern. *`

File: tests/Kernel/CorporateBlocksFooterTest.php [hunk 10]
```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\Kernel;
+
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Class CorporateBlocksFooterTest.
```
Comment (ademarco) [inline tests/Kernel/CorporateBlocksFooterTest.php]: Add meaningful class description.

File: tests/Kernel/CorporateBlocksFooterTest.php [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+  public function testFooterBlockRendering(): void {
+    // Setup and render footer block.
+    $config = [
+      'id' => 'oe_footer',
+      'label' => 'OpenEuropa footer block',
+      'provider' => 'oe_corporate_blocks',
+      'label_display' => '0',
+    ];
+
+    $render = $this->buildBlock('oe_footer', $config);
```
Comment (ademarco) [inline tests/Kernel/CorporateBlocksFooterTest.php]: We should override here the block footer configuration object by providing test configuration, so we can make sure that the template styled the footer correctly. I propose to have just 2 links per link category and test labels.

File: tests/Kernel/CorporateBlocksFooterTest.php [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+    $crawler = new Crawler($html);
+
+    // Make sure that footer block is present.
+    $actual = $crawler->filter('footer.ecl-footer');
+    $this->assertCount(1, $actual);
+    // Make sure that footer block rendered correctly.
+    $actual = $crawler->filter('footer.ecl-footer div.ecl-footer__corporate-top div.ecl-row div.ecl-footer__column');
+    $this->assertCount(3, $actual);
+    $actual = $crawler->filter('footer.ecl-footer div.ecl-footer__corporate-bottom div.ecl-row ul.ecl-footer__list li.ecl-footer__list-item');
+    $this->assertCount(7, $actual);
```
Comment (ademarco) [inline tests/Kernel/CorporateBlocksFooterTest.php]: Following the comment above we should add here assertions that assert links and labels are styled correctly.
